src/components/dialogs/AddActivityDialog.js

```diff
@@ -80,28 +90,26 @@ class ActivityDialog extends React.PureComponent {
       });
     }
   }
-  handelfetchGithubFilesStatus=(fetchGithubFilesStatus,activity)=>{
-    switch(fetchGithubFilesStatus){
-      case "SUCCESS":{
+
+  handelfetchGithubFilesStatus = (fetchGithubFilesStatus,activity) => {
+    switch (fetchGithubFilesStatus) {
+      case "SUCCESS":
         this.fetchedGithubURL = this.state.githubURL;
         this.setState({ files : activity.files || []})
         this.hideLoading();
         break;
-      }
-      case "LOADING":{
+      case "LOADING":
         this.showLoading();
         break;
-      }
-      case "ERROR":{
+      case "ERROR":
         this.hideLoading();
         break;
-      }
-      default : {
-        break;
-      }
+      default:
+        this.hideLoading();
     }
   }
-  getTypeSpecificElements() {
```

Comment:
@Brainenjii Just curious, we didn't bind `getTypeSpecificElements()` method anywhere in the code, how were we able to call `this.getTypeSpecificElements()` in this component previously?

Reply:
`getTypeSpecificElements` is method of `ActivityDialog` class (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods). So, it's able to be invoked with this from any other method of this class

Reply:
i get it now. we are not passing that method as props or event-handler to another component in this case. thanks!